hera_pspec/uvwindow.py

```diff
@@ -0,0 +1,1320 @@
+from pyuvdata import utils as uvutils
+import uvtools.dspec as dspec
+import h5py
+import warnings
+import numpy as np
+import sys
+import os
+import copy
+import time
+from scipy.interpolate import interp2d
+from pyuvdata import UVBeam, UVData
+from astropy import units
+
+from . import conversions, noise, version, pspecbeam, grouping, utils
+from . import uvpspec_utils as uvputils
+
+
+class FTBeam():
+    """Class for :class:`FTBeam` objects."""
+
+    def __init__(self, pol, spw_range=None, ftfile=None,
+                 verbose=False, x_orientation=None):
+        """
+        Obtain Fourier transform of beam in sky plane for given frenquencies.
+
+        Given a polarisation pair and a spectral window,
+        read beam simulations to get the Fourier transform of the instrument
+        beam in the sky plane for all the frequencies in the spectral
+        window. Output is an array of dimensions (kperpx, kperpy, freq).
+        Computations correspond to the Fourier transform performed in equation 
+        10 of memo.
+
+        Parameters
+        ----------
+        pol : str or int
+            Can be pseudo-Stokes or power: 
+            in str form: 'pI', 'pQ', 'pV', 'pU', 'xx', 'yy', 'xy', 'yx'
+            in number form: 1, 2, 4, 3, -5, -6, -7, -8
+        spw_range : tuple or array
+            In (start_chan, end_chan). Must be between 0 and 1024 (HERA
+            bandwidth).
+            If None, whole instrument bandwidth is considered.
+        ftfile : str
+            Access to the Fourier transform of the beam on the sky plane
+            (Eq. 10 in Memo)
+            Options are;
+                - Load from file. Then input is the root name of the file
+                to use, without the polarisation
+                Ex : ft_beam_HERA_dipole (+ path)
+                - None (default). Computation from beam simulations (slow).
+                Not yet implemented.
+        verbose : bool, optional
+            If True, print progress, warnings and debugging info to stdout.
+        x_orientation: str, optional
+            Orientation in cardinal direction east or north of X dipole.
+            Default keeps polarization in X and Y basis.
+            Used to convert polstr to polnum and convertly.
+        """
+        # verbose
+        try:
+            bool(verbose)
+        except ValueError:
+            raise ValueError("verbose must be boolean")
```

Comment:
Actually I ended up choosing this rather convoluted `try` statement because `bool(verbose)` very rarely raises an error. It basically interpret anything (let's say any reasonable input such as an array, a string, etc.) that is not zero as `True`... Do you think that is sufficient?

Reply:
Yes, most things are convertible to bools in python, which is fine. But if `bool(verbose)` rarely raises an error, you're not achieving anything more by putting it in a try/except -- it still won't raise the error! You *could* do something like `if not isinstance(verbose, bool)` but I would *definitely* not recommend this. Just document that it should be a bool and convert it to such.